1/main.rs

```diff
@@ -0,0 +1,34 @@
+// Parse standard in as a sequence of integers.
+fn input_ints() -> Vec<i32> {
+	use std::io::prelude::*;
+	std::io::stdin().lock().lines().map(
+		|line| line.unwrap().trim().parse::<i32>().unwrap()
+	).collect()
+}
+
+// Sum inputs in windows of 3.
+fn boxcar_sum(signal : Vec<i32>) -> Vec<i32> {
+    let mut result = Vec::new();
```

Comment:
Is there a way to pre-allocate capacity in this vec? I know what the size will be in advance; it seems silly to wait for the stdlib to resize it dozens of times

Reply:
Yep! https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity